lib/src/optimism/mod.rs

```diff
@@ -236,119 +282,217 @@ impl<D: BatcherDb> DeriveMachine<D> {
                     self.op_block_seq_no = 0;
                     self.op_batcher.state.do_next_epoch()?;
 
-                    self.op_batcher
-                        .state
-                        .epoch
-                        .deposits
-                        .iter()
-                        .map(|tx| tx.to_rlp())
-                        .collect()
+                    self.op_batcher.state.epoch.deposits.clone()
                 } else {
                     self.op_block_seq_no += 1;
 
-                    Vec::new()
+                    vec![]
                 };
 
-                // obtain verified op block header
-                let new_op_head = {
-                    // load the next op block header
-                    let new_op_head = self
-                        .derive_input
-                        .db
-                        .get_op_block_header(self.op_block_no)
-                        .context("op block not found")?;
-
-                    // Verify that the op block header loaded from the DB matches the payload
-                    // attributes of the batch.
-                    ensure!(
-                        new_op_head.parent_hash == self.op_batcher.state.safe_head.hash,
-                        "Invalid op block parent hash"
-                    );
-                    ensure!(
-                        new_op_head.beneficiary == self.op_batcher.config.sequencer_fee_vault,
-                        "Invalid op block beneficiary"
-                    );
-                    ensure!(
-                        new_op_head.gas_limit == self.op_batcher.config.system_config.gas_limit,
-                        "Invalid op block gas limit"
-                    );
-                    ensure!(
-                        new_op_head.timestamp == U256::from(op_batch.0.timestamp),
-                        "Invalid op block timestamp"
-                    );
-                    ensure!(
-                        new_op_head.extra_data.is_empty(),
-                        "Invalid op block extra data"
+                let l1_epoch_header_mix_hash = self
+                    .derive_input
+                    .db
+                    .get_full_eth_block(op_batch.0.epoch_num)
+                    .context("eth block not found")?
+                    .block_header
+                    .mix_hash;
+
+                // From the spec:
+                // The first transaction MUST be a L1 attributes deposited transaction,
+                // followed by an array of zero-or-more user-deposited transactions.
+                let l1_attributes_tx = self.derive_l1_attributes_deposited_tx(&op_batch);
+                // TODO: revise that skipping undecodable transactions is part of spec
```

Comment:
Important point, we might want to check what OP-geth is doing with invalid transactions

Reply:
OK, I did some research on this:
- This is analog to Ethereum, the CL creates the `PayloadAttribute` and passes it to the EL to create the `ExecutionPayload`, essentially what we call block input (maybe we should consider renaming)
- The `ExecutionPayload` contains the RLP-encoded transactions, if they cannot be decoded the block is invalid.
- For OP the `PayloadAttribute` also contains transactions, that must then be included by the EL.
- If the `PayloadAttribute` contains invalid transactions, the entire request is invalid and we essentially have a consensus failure.
- I am not sure what mechanisms are there to recover from such a failure.

This means that undecodable transactions are **not** skipped, but they must all be valid.

However, I guess this means that we do not really need the RLP-decoding and could instead use the approach from before that downloads the transactions from RPC and just checks that the trie hash is correct. This could simplify our code dramatically.

Reply:
With this in mind, I am wondering wether we actually need to prove the block failure... This would also be some kind of consensus error and I don't think that skipping to the next batch is the designed behavior for such a case. Maybe we should just leave it at stopping at consensus errors.

Reply:
Funnily enough, op-geth neither returns a state reset nor an invalid attribute error, so i guess it would just stop in this case.

Reply:
there could be other problems with the block that could lead to dropping its batch, which needs to be justified with a proof.